src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = MarketItemType.SUPPLY.toString().toLowerCase();
+    private static final String BUY_LABEL = MarketItemType.BUY.toString().toLowerCase();
+    private static final String RESULT_LABEL = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        MarketData marketData = new MarketData();
+
+        readFromFile(fromFileName, (line) -> {
+            String[] lineData = line.split(",");
+            String type = lineData[0].toLowerCase().trim();
+            int value = Integer.parseInt(lineData[1].trim());
+            if (type.equals(SUPPLY_LABEL)) {
+                marketData.addSupplyTotal(value);
+            } else if (type.equals(BUY_LABEL)) {
+                marketData.addBuyTotal(value);
+            }
+        });
+
+        String statistic = getStatisticString(marketData);
+        writeStatisticToFile(toFileName, statistic);
+
+        return statistic;
+    }
+
+    private void readFromFile(String fileName, LineProcessor processor) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                processor.execute(line);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read data from the file " + fileName, e);
+        }
```

Lines 35-43 (readFromFile catch): You rethrow a RuntimeException with a generic message when any exception occurs reading/parsing. To be more helpful and to follow "#### Don't ignore exceptions", include the problematic line (and optionally line number) in the error message when the failure is caused by malformed content. Consider doing parsing/validation in `readFromFile` (where `fileName` and a line counter are available) so you can throw a descriptive exception like `new RuntimeException("Malformed line in " + fileName + ": '" + line + "'", e)` .